### PR TITLE
Upgrade dependencies to allow us to use SQLAlchemy v2 in Airflow 3.0/main

### PR DESCRIPTION
**Pull request by ashb**

### CODE REVIEW

**Comment by ashb** [comment]
This failed before once landed in main https://github.com/apache/airflow/pull/42898 -- but lets try again/more.

File: generated/provider_dependencies.json [hunk 1]
```diff
@@ -38,7 +38,6 @@
       "jsonpath_ng>=1.5.3",
       "python3-saml>=1.16.0",
       "redshift_connector>=2.0.918",
-      "sqlalchemy_redshift>=0.8.6",
```
Comment (vincbeck) [inline generated/provider_dependencies.json:41]: I tried to remove it in #42830 but it seems some tests failed (but after the merge, which was really weird). We reverted it then

Comment (ashb) [reply generated/provider_dependencies.json:41]: Yeah I was aware and going to try poking at it again

Comment (vincbeck) [reply generated/provider_dependencies.json:41]: Cool :D Godspeed!

File: providers/src/airflow/providers/fab/auth_manager/models/__init__.py [hunk 2]
```diff
@@ -58,12 +58,25 @@
...
+        :meta private:
+        """
+
+        metadata = metadata
+        # https://docs.sqlalchemy.org/en/20/changelog/migration_20.html#migration-20-step-six
+        __allow_unmapped__ = True
+
     Model.metadata = metadata
+
+    flask_appbuilder.models.sqla.Model = Model
```
Comment (ashb) [inline providers/src/airflow/providers/fab/auth_manager/models/__init__.py:79]: This is almost certainly horribly horribly wrong.

**Pull request closed without merging**